### PR TITLE
Add content-type/preview_image API

### PR DESCRIPTION
**Pull request by phuongnq**
### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/5759

### CODE REVIEW

File: src/main/api/2/studio-api-2.yaml [hunk 1]
```diff
@@ -3488,6 +3488,52 @@ paths:
...
+      summary: Get preview image of a given content-type
+      operationId: getContentTypePreviewImage
+      parameters:
+        - name: siteId
+          in: query
+          description: site ID
+          required: true
+          schema:
+            type: string
+        - name: contentType
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: `contentTypeId`

File: src/main/api/2/studio-api-2.yaml [hunk 2]
```diff
@@ -3488,6 +3488,52 @@ paths:
         '403':
           $ref: '#/components/responses/Forbidden'
 
+  /configuration/content-type/preview_image:
+    get:
+      tags:
+        - configuration
+      summary: Get preview image of a given content-type
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: Get preview image of a given content type

File: src/main/java/org/craftercms/studio/controller/rest/v2/ConfigurationController.java [hunk 3]
```diff
@@ -160,6 +168,20 @@ public ResponseBody getContentTypeUsage(@RequestParam String siteId, @RequestPar
         return body;
     }
 
+    @GetMapping("content-type/preview_image")
+    public ResponseEntity<Resource> getContentTypePreviewImage(@RequestParam String siteId, @RequestParam String contentType)
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/controller/rest/v2/ConfigurationController.java]: contentTypeId

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml]: contentTypeId

File: src/main/api/2/studio-api-2.yaml [hunk 4]
```diff
@@ -3488,6 +3488,52 @@ paths:
...
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: 'OK'
+          content:
+            '*/*':
+              schema:
+                type: string
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:3517]: Is this the correct type?

File: src/main/api/2/studio-api-2.yaml [hunk 5]
```diff
@@ -3488,6 +3488,52 @@ paths:
         '403':
           $ref: '#/components/responses/Forbidden'
 
+  /configuration/content-type/preview_image:
+    get:
+      tags:
+        - configuration
+      summary: Get preview image of a given content-type
+      operationId: getContentTypePreviewImage
```
Comment (sumerjabri) [inline src/main/api/2/studio-api-2.yaml:3498]: Add `description` and indicate permissions.

Something like:
`Required permission "read_configuration"`

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java [hunk 6]
```diff
@@ -196,4 +221,25 @@ protected String getContentTypePath(String contentType) {
         return normalize(contentTypeBasePathPattern.replaceFirst("\\{content-type}", contentType));
     }
 
+    protected String getContentTypePreviewImageFilename(String siteId, String contentTypeId) throws ServiceLayerException {
+        if (!siteService.exists(siteId)) {
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java]: You can just use
siteService.checkSiteExists(siteId);

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (phuongnq) [reply src/main/api/2/studio-api-2.yaml]: Updated. Thank you!

File: src/main/api/2/studio-api-2.yaml (hunk 2 above)
Comment (phuongnq) [reply src/main/api/2/studio-api-2.yaml]: Updated. Thank you!

File: src/main/java/org/craftercms/studio/controller/rest/v2/ConfigurationController.java (hunk 3 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/controller/rest/v2/ConfigurationController.java]: Updated. Thank you!

File: src/main/api/2/studio-api-2.yaml (hunk 1 above)
Comment (phuongnq) [reply src/main/api/2/studio-api-2.yaml]: Updated. Thank you!

File: src/main/api/2/studio-api-2.yaml (hunk 5 above)
Comment (phuongnq) [reply src/main/api/2/studio-api-2.yaml:3498]: Updated. Thank you!

File: src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java (hunk 6 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java]: Updated. Thank you!

File: src/main/api/2/studio-api-2.yaml (hunk 4 above)
Comment (phuongnq) [reply src/main/api/2/studio-api-2.yaml:3517]: @sumerjabri We only want to respond the binary image, right? If it is true, the type here is correct. The response is a string in binary format (same as in https://app.swaggerhub.com/apis/craftercms/studio/4.0.1.0#/plugin/getPluginFile )

**Review by jmendeza — APPROVED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java [hunk 7]
```diff
@@ -53,6 +55,19 @@ public ContentTypeUsage getContentTypeUsage(String siteId, String contentType) t
...
+    /**
+     * Finds the preview image for a given content-type
+     *
+     * @param siteId the id of the site
+     * @param contentType the id of the content-type
+     * @return the preview image file as a pair of path and resource
+     * @throws ServiceLayerException if there is any error finding the items
+     */
+    @Override
+    public ImmutablePair<String, Resource> getContentTypePreviewImage(String siteId, String contentType) throws ServiceLayerException {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java]: contentTypeId

File: src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java [hunk 8]
```diff
@@ -139,6 +150,20 @@ public ContentTypeUsage getContentTypeUsage(String siteId, String contentType) t
         return usages;
     }
 
+    @Override
+    public ImmutablePair<String, Resource> getContentTypePreviewImage(String siteId,
+                                                                      @ValidateSecurePathParam(name = "contentType") String contentType) throws ServiceLayerException {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java]: "contentTypeId" and contentTypeId

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java [hunk 9]
```diff
@@ -51,6 +56,7 @@ public ContentTypeServiceImpl(ContentTypeServiceInternal contentTypeServiceInter
      * @throws ServiceLayerException if there is any error finding the items
      */
     @Override
+    @HasPermission(type = DefaultPermission.class, action = PERMISSION_CONTENT_READ)
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java:61]: Remember to use @HasPermission in combination with @ProtectedResourceId when the permission is site or path dependent. e.g.: https://github.com/craftercms/studio/blob/develop/src/main/java/org/craftercms/studio/impl/v2/service/content/ContentServiceImpl.java#L103

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java (hunk 7 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java]: Fixed. Thank you!

File: src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java (hunk 8 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/content/internal/ContentTypeServiceInternalImpl.java]: Fixed. Thank you!

File: src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java (hunk 9 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/content/ContentTypeServiceImpl.java:61]: Got it. Thanks for the hint!

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**